app/models/events/search.rb

```diff
@@ -21,32 +23,34 @@ class Search
     before_validation { self.distance = nil if distance.blank? }
     before_validation(unless: :distance) { self.postcode = nil }
 
-    def available_event_types
-      @available_event_types ||= GetIntoTeachingApiClient::Constants::EVENT_TYPES.map do |key, value|
-        GetIntoTeachingApiClient::TypeEntity.new(id: value, value: key)
+    class << self
+      def available_event_types
+        @available_event_types ||= GetIntoTeachingApiClient::Constants::EVENT_TYPES.map do |key, value|
+          GetIntoTeachingApiClient::TypeEntity.new(id: value, value: key)
+        end
       end
-    end
 
-    def available_event_type_ids
-      available_event_types.map(&:id)
-    end
+      def available_event_type_ids
+        available_event_types.map(&:id)
+      end
 
-    def available_distances
-      [["Nationwide", nil]] + DISTANCES.map { |d| ["Within #{d} miles", d] }
-    end
+      def available_distance_keys
+        available_distances.map(&:last)
+      end
 
-    def available_distance_keys
-      available_distances.map(&:last)
-    end
+      def available_distances
+        [["Nationwide", nil]] + DISTANCES.map { |d| ["Within #{d} miles", d] }
+      end
 
-    def available_months
-      (0..5).map do |i|
-        month = i.months.from_now.to_date
+      def available_months
```

Comment:
I guess this could be shortened a bit to something like `(0..5).map { |m| m.months.from_now.strftime("%Y-%m") }`, but no strong feelings